dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresConstants.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+internal static class PostgresConstants
+{
+    /// <summary>A <see cref="HashSet{T}"/> of types that a key on the provided model may have.</summary>
+    public static readonly HashSet<Type> SupportedKeyTypes =
+    [
+        typeof(short),
+        typeof(int),
+        typeof(long),
+        typeof(string),
+        typeof(Guid),
+    ];
+
+    /// <summary>A <see cref="HashSet{T}"/> of types that data properties on the provided model may have.</summary>
+    public static readonly HashSet<Type> SupportedDataTypes =
+    [
+        typeof(bool),
+        typeof(bool?),
+        typeof(short),
+        typeof(short?),
+        typeof(int),
+        typeof(int?),
+        typeof(long),
+        typeof(long?),
+        typeof(float),
+        typeof(float?),
+        typeof(double),
+        typeof(double?),
+        typeof(decimal),
+        typeof(decimal?),
+        typeof(string),
+        typeof(DateTimeOffset),
+        typeof(DateTimeOffset?),
+        typeof(Guid),
+        typeof(Guid?),
+        typeof(byte[]),
+        typeof(List<bool>),
```

Comment:
Did you consider supporting more enumerables? There are a lot of helpers to make this easier to support.  E.g.
```csharp
// supportEnumerable checks for any enumerable versions of the supperted types if true
public void VectorStoreRecordPropertyReader.VerifyDataProperties(HashSet<Type> supportedTypes, bool supportEnumerable)

// This can create the required output enumerable from the input data. E.g. if the db returns an array or list, but you need a ICollection, it can create the right type.
VectorStoreRecordMapping.CreateEnumerable<T>(IEnumerable<T> input, Type requiredEnumerable)
```